### PR TITLE
UCT/API: remove redundant UCT_CB_FLAG_SYNC

### PR DESCRIPTION
**Pull request by evgeny-leksikov**
## What
Remove redundant UCT_CB_FLAG_SYNC

## Why ?
Flags UCT_CB_FLAG_SYNC and UCT_CB_FLAG_ASYNC are mutually exclusive

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7912/ for details (Mellanox internal link).

**Comment by shamisp** [comment]
@hjelmn FYI

**Comment by hjelmn** [comment]
Yeah, was wondering about that. Seemed odd to have both.

File: src/uct/api/uct.h [hunk 1]
```diff
@@ -338,22 +338,21 @@ enum uct_msg_flags {
...
+                                         invocation, the interface must have the
+                                         @ref UCT_IFACE_FLAG_CB_ASYNC flag set.
+                                         If async callback is requested on an
+                                         interface which only supports sync
+                                         callback (i.e., only the @ref
+                                         UCT_IFACE_FLAG_CB_SYNC flag is set),
+                                         the callback may be invoked only from
+                                         the context that called @ref
+                                         uct_iface_progress). */
 };
```
Comment (hjelmn) [inline src/uct/api/uct.h]: Its probably a bad idea to change the UCS_BIT of the async flag. Can break code that is using UCT. Probably better off just blocking out UCS_BIT(1) for awhile and leaving async as UCS_BIT(2).

**Comment by shamisp** [comment]
@hjelmn technically it breaks backward compatibility. But if you are not concerned about this, I'm not supper worried.

**Comment by hjelmn** [comment]
@shamisp I am not too concerned about it. Just concerned at the possibility that UCS_BIT(1) gets redefined at some point to something I don't want. Though I don't use ASYNC right now so I guess just protecting myself by checking for the sync flag in btl/uct is enough.

**Comment by shamisp** [comment]
@hjelmn - maybe we should start with UCS_BIT(3) ? @yosefe ?

**Comment by evgeny-leksikov** [comment]
@shamisp why it's better than UCS_BIT(0) which was not used as well? may be rename UCS_BIT(1) and UCS_BIT(2) to UCT_CB_FLAG_DEPRECATED_XXX until lib version is bumped?

**Comment by hjelmn** [comment]
@evgeny-leksikov Just doesn't make as much sense to change the value of the async flag. Then could also block out the old sync flag until the so is bumped. That would make sense to me.

**Comment by shamisp** [comment]
@hjelmn - agreed. This essentially was my intend.

**Comment by evgeny-leksikov** [comment]
So, just to clarify:
1) UCT_CB_FLAG_ASYNC = UCS_BIT(2) - old value
2) return UCS_ERR_INVALID_PARAM if UCS_BIT(1) is set or block it by some "deprecated" name?

**Comment by hjelmn** [comment]
@evgeny-leksikov Yes. That would be the safest approach. But do both the block and error return to be safe.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7945/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7954/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7957/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@shamisp @gmegan @hjelmn pls take a look

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7960/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7962/ for details (Mellanox internal link).

**Comment by shamisp** [comment]
👍

**Comment by evgeny-leksikov** [comment]
Not related failures should be fixed by https://github.com/openucx/ucx/pull/2958
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7969/ for details (Mellanox internal link).

**Comment by evgeny-leksikov** [comment]
timed out
bot:mlx:retest

**Comment by evgeny-leksikov** [comment]
#2958
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7979/ for details (Mellanox internal link).

**Comment by evgeny-leksikov** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7982/ for details (Mellanox internal link).

**Comment by evgeny-leksikov** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7987/ for details (Mellanox internal link).

**Comment by evgeny-leksikov** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7995/ for details (Mellanox internal link).

**Pull request merged**